### ISSUE TITLE
Improve EPUB 2 OPF guide

### ISSUE THREAD

**Issue by pedrosanta** [cyrilis/epub-gen#16]
Hi all!

Just firing up this issue, as a note to improve the `guide` part of OPF template for EPUB 2, and to use `landmarks` for EPUB 3, as the previous is deprecated. 😉 

More info about OPF `guide` section here http://www.idpf.org/epub/20/spec/OPF_2.0.1_draft.htm#Section2.6 and here http://www.idpf.org/epub/30/spec/epub30-publications.html#sec-guide-elem.

**Comment by cyrilis** [comment]
Thanks @pedrosanta ,

Current OPF file template is a little some chaos, will improve it later. 

BTW, what do you think of current option structure and `Table of content (TOC)` template? If a flat array of html is enough for a epub structure? or should I change it to embedded array to support deeper levels？

**Comment by pedrosanta** [comment]
Yep, don't really worry about this issue, I just created it to note an improvement that can be made to the EPUB 2 OFP guide part (which is blank I think), but, really as a nice to have, as... `guide` has been deprecated in EPUB 3 (replaced by landmarks, that fall on the xhtml TOC). So, just leaving the issue as a note, actually.

Regarding the structure of TOC, I think it's OK at the moment. On the Reedsy book exporter we do have a TOC with deeper levels (which I build using a mix of adding a `headings` property/array to the content and using that on custom templates).

But it's something I'm willing to discuss. An important thing we have to be in mind is... for each OPF content (which best practices say it should be on several HTMLs to improve readers content loading performance, exactly like the lib is doing, so OK on that part) I'm pretty sure the readers will break to a new page. (So using nested content for headings, in my case, wouldn't work.)

But I think this could fall on another issue, focused in this subject if we want to discuss it further. 😀  So given that's not a big need from my end, and if you don't also have that need, I would only consider if I got a very strong need/ask for a nested structure, IMHO.

PS: A thing that I feel the need atm, actually, is being able to customize the TOC that's generated further, as in, where to place it - in my example I add a cover page, a blank page, a book title page, a publishing info page, and only after these, the TOC. I kinda omitted the TOC on my custom OPF, and generated a new one manually and added it as regular content, but then validation bitches about having an `.xhtml` file that's not declared on the OPF. But regarding that I'll think about it, submit a PR or open an issue.

Cheers @cyrilis! 🚀

**Comment by pedrosanta** [comment]
Ok, I'm opening an issue to discuss a bit more advanced TOC customization - to keep conversations/issues tidy. 😉
